### PR TITLE
Add increment / decrement buttons to text boxes with number mask

### PR DESCRIPTION
**Pull request by BitwiseAndrea**
- Modify Number mask verify method to return false on empty string, so that the textbox always has a valid number in it (this is an explicit change in behavior and might not be the right solution)
- Add Increment and Decrement buttons to Defaults
    - Styling and structure copied from Dropdown button 
- Add Increment and Decrement buttons to self.Frame when mask type is set to Number
    - Not sure if theres a better way to check than == "Number"
    - Buttons are visually inset, might be concerns with overflow of large numbers
    - The buttons being nested inside the textbox makes for... odd focus pulling? Not sure how to fix it in a reasonable way. Might just be a matter of not having the buttons inset and then resizing the container frame to shrink a bit and give space for the buttons.
- Clean up buttons if they exist on destroy

TODO: Add up/down arrow key support - possibly behind flag, definitely requires ClearTextOnFocus to be false

https://user-images.githubusercontent.com/50600470/103394369-1fbfb600-4af6-11eb-846a-b622866b51a5.mov

### CODE REVIEW

File: Source/Classes/TextMask.lua [hunk 1]
```diff
@@ -105,10 +109,36 @@ function TextMaskClass:SetMaskType(name)
...
+
+		self.incrementButton = INCREMENT_BUTTON:Clone()
+		self.incrementButton.Position = UDim2.new(0, self.Frame.AbsoluteSize.X - self.Frame.AbsoluteSize.Y / 2, 0, 0)
+		self.incrementButton.Activated:Connect(function()
+			self.Frame.Text = tonumber(self.Frame.Text) + 1
+		end)
+		self.incrementButton.Parent = self.Frame
+		
+		self.decrementButton = DECREMENT_BUTTON:Clone()
+		self.decrementButton.Position = UDim2.new(0, self.Frame.AbsoluteSize.X - self.Frame.AbsoluteSize.Y / 2, 0, self.Frame.AbsoluteSize.Y / 2)
```
Comment (BitwiseAndrea) [inline Source/Classes/TextMask.lua:126]: ok i can't trust absolute size so i'm going to figure out a better way to do this

File: Source/Classes/TextMask.lua [hunk 2]
```diff
@@ -105,10 +109,36 @@ function TextMaskClass:SetMaskType(name)
...
 	self._MaskType = mask
 	self.Frame.Text = mask:Process(self.Frame.Text):sub(1, self._MaxLength)
+	
+	-- Add increment/decrement buttons for numeric inputs
+	if name == "Number" then
+		-- Initialize text to 0, consumer can override
+		self.Frame.Text = 0
+
+		self.incrementButton = INCREMENT_BUTTON:Clone()
+		self.incrementButton.Position = UDim2.new(0, self.Frame.AbsoluteSize.X - self.Frame.AbsoluteSize.Y / 2, 0, 0)
```
Comment (EgoMoose) [inline Source/Classes/TextMask.lua:119]: Problems could arrise from using absolute size in this context. If the user decided to resize `self.Frame` then the scaling will not carry over.

The same effect can be achieved w/ scale and anchor points.

```suggestion
self.IncrementButton.AnchorPoint = Vector2.new(1, 0)
self.IncrementButton.Position = UDim2.new(1, 0, 0, 0)
```

In addition for sake of matching the rest of the code base the `i` in `increment` should be capitalized.

File: Source/Classes/TextMask.lua (hunk 1 above)
Comment (EgoMoose) [reply Source/Classes/TextMask.lua:126]: Similaliry to the comment above:

```suggestion
self.DecrementButton.AnchorPoint = Vector2.new(1, 1)
self.DecrementButton.Position = UDim2.new(1, 0, 1, 0)
```

File: Source/Classes/TextMask.lua [hunk 3]
```diff
@@ -105,10 +109,36 @@ function TextMaskClass:SetMaskType(name)
...
+		self.incrementButton.Activated:Connect(function()
```
Comment (EgoMoose) [inline Source/Classes/TextMask.lua:120]: This event should be marked by the maid. There is the possibility that a user will reparent these increment buttons and as such using `:Destroy()` on the parent will not properly disconnect this event.

File: Source/Classes/TextMask.lua [hunk 4]
```diff
@@ -105,10 +109,36 @@ function TextMaskClass:SetMaskType(name)
...
+			self.Frame.Text = tonumber(self.Frame.Text) + 1
+		end)
+		self.incrementButton.Parent = self.Frame
+		
+		self.decrementButton = DECREMENT_BUTTON:Clone()
+		self.decrementButton.Position = UDim2.new(0, self.Frame.AbsoluteSize.X - self.Frame.AbsoluteSize.Y / 2, 0, self.Frame.AbsoluteSize.Y / 2)
+		self.decrementButton.Activated:Connect(function()
```
Comment (EgoMoose) [inline Source/Classes/TextMask.lua:127]: See my comment on maid and events above.

File: Source/Classes/TextMask.lua [hunk 5]
```diff
@@ -105,10 +109,36 @@ function TextMaskClass:SetMaskType(name)
...
+			self.Frame.Text = tonumber(self.Frame.Text) - 1
+		end)
+		self.decrementButton.Parent = self.Frame
+	end
 end
 
 function TextMaskClass:Destroy()
 	self._Maid:Sweep()
+	if self.IncrementButton then
```
Comment (EgoMoose) [inline Source/Classes/TextMask.lua:136]: These could be captured by the maid when they're created, no need to manually check and destroy.

**Review by EgoMoose — APPROVED** [review]
I left a few comments/feedback if your interested (I'll implement the changes myself) but otherwise it lgtm!

Thanks again for contributing!

File: Source/Classes/TextMask.lua (hunk 5 above)
Comment (EgoMoose) [reply Source/Classes/TextMask.lua:136]: Also the capitalization here is inconsistent.

**Comment by EgoMoose** [comment]
Changes can be found in this commit:
ccaf2914328cbbbe13b02ce70f43b80598b491d7

**Pull request merged**